Tests/Integration/IndexQueue/FrontendHelper/PageIndexerTest.php

```diff
@@ -285,67 +287,133 @@ public function canExecuteAdditionalPageIndexer(): void
         $this->addTypoScriptToTemplateRecord(1, 'config.index_enable = 1');
-        $this->indexQueuedPage(24, '/en/', ['MP' => '24-14']);
+        $this->indexQueuedPage();
 
         // we wait to make sure the document will be available in solr
         $this->waitToBeVisibleInSolr();
 
         $solrContent = file_get_contents($this->getSolrConnectionUriAuthority() . '/solr/core_en/select?q=*:*');
-        self::assertStringContainsString('"title":"FirstShared (Not root)"', $solrContent, 'Could not find content from mounted page in solr');
+        self::assertStringContainsString('"title":"FirstShared"', $solrContent, 'Could not find content from mounted page in Solr');
     }
 
     /**
+     * This testcase should check if we can index a cross-site mount, a mount point
+     * that mounts a page from another site
+     *
      * There is following scenario:
      *
      *  [0]
      *  |
-     *  ——[20] Shared-Pages (Not root)
+     *  |—[ 111] Page (Root Testpage 2)
      *  |   |
-     *  |   ——[44] FirstShared (Not root)
+     *  |   |—[24] FirstShared
+     *  |
+     *  |—[ 1] Page (Root Testpage 1)
+     *      |—[14] Mount Point (to [24] to show contents from)
+     *
+     *
+     *  Note: Indexing a cross-site mount point requires that "index_enable" is set
+     *  in the pagetree of the mounted page, as TYPO3 loads the TypoScript
+     *  of the mounted page since TYPO3 13.
+     */
+    #[Test]
+    public function canIndexCrossSiteMounts(): void
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['FE']['enable_mount_pids'] = 1;
+
+        $this->cleanUpAllCoresOnSolrServerAndAssertEmpty();
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/can_index_mounted_page_from_another_site.csv');
+        $this->addTypoScriptToTemplateRecord(1, 'config.index_enable = 1');
+        $this->addTypoScriptToTemplateRecord(111, 'config.index_enable = 1');
+        $this->indexQueuedPage();
+
+        // we wait to make sure the document will be available in solr
+        $this->waitToBeVisibleInSolr();
+
+        $solrContent = file_get_contents($this->getSolrConnectionUriAuthority() . '/solr/core_en/select?q=*:*');
+        self::assertStringContainsString('"title":"FirstShared"', $solrContent, 'Could not find content from mounted page in Solr');
+    }
+
+    /**
+     * This testcase should check if we can index a cross-site mount, a mount point
+     * that mounts a page from another pagetree that has no site configuration
+     *
+     * There is following scenario:
+     *
+     *  [0]
      *  |
-     *  ——[ 1] Page (Root)
+     *  |—[ 20]  Shared-Pages (Not root)
+     *  |   |
+     *  |   |—[24] FirstShared (Not root)
      *  |
-     *  ——[14] Mount Point (to [24] to show contents from)
+     *  |—[ 1] Page (Root Testpage 1)
+     *      |
+     *      |—[14] Mount Point (to [24] to show contents from)
+     */
+    #[Test]
+    public function canIndexCrossSiteMountsWithoutSiteConfiguration(): void
```

Comment:
Is that scenario possible in TYPO3 Backend at all? 
As far as I know, TYPO3 creates a site-config.yaml file as soon the page is marked as root.
Can that test-case be removed?

Reply:
This is the original test using a sysfolder with pages that should be mounted (no site root), but currently TYPO3 cannot mount such pages and an exception occurs:

```
 (1/1) #1521716622 TYPO3\CMS\Core\Exception\SiteNotFoundException 
No site found in root line of page 86
```

This issue already exists in TYPO3 12, I am currently not sure why this test worked in TYPO3 12, test results in v12 might be misleading.

As it's not a valid and supported scenario in the TYPO3 core, we could IMHO delete the test.

Reply:
I kept this test for now, till we could check if this is a bug in TYPO3 or if mounts outside of a side aren't intended to work